src/overlays/actors/ovl_Obj_Takaraya_Wall/z_obj_takaraya_wall.c

```diff
@@ -30,39 +50,450 @@ ActorInit Obj_Takaraya_Wall_InitVars = {
     (ActorFunc)ObjTakarayaWall_Draw,
 };
 
-// static ColliderCylinderInit sCylinderInit = {
-static ColliderCylinderInit D_80ADA2A0 = {
-    { COLTYPE_NONE, AT_NONE, AC_NONE, OC1_ON | OC1_TYPE_ALL, OC2_TYPE_2, COLSHAPE_CYLINDER, },
-    { ELEMTYPE_UNK0, { 0x00000000, 0x00, 0x00 }, { 0x00000000, 0x00, 0x00 }, TOUCH_NONE | TOUCH_SFX_NORMAL, BUMP_NONE, OCELEM_ON, },
+static ColliderCylinderInit sCylinderInit = {
+    {
+        COLTYPE_NONE,
+        AT_NONE,
+        AC_NONE,
+        OC1_ON | OC1_TYPE_ALL,
+        OC2_TYPE_2,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK0,
+        { 0x00000000, 0x00, 0x00 },
+        { 0x00000000, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_NONE,
+        OCELEM_ON,
+    },
     { 40, 120, 0, { 0, 0, 0 } },
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80ADA2CC[] = {
+static InitChainEntry sInitChain[] = {
     ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
 };
 
-#endif
+#define TAKARAYA_WALL_ROWS 11
+#define TAKARAYA_WALL_COLUMNS 8
+
+typedef struct ObjTakarayaWallSpace {
+    /* 0x0 */ u8 row;
+    /* 0x1 */ u8 column;
+} ObjTakarayaWallSpace; // size = 0x2
+
+typedef enum ObjTakarayaWallSpaceState {
+    /* 0 */ TAKARAYA_WALL_INACTIVE,
+    /* 1 */ TAKARAYA_WALL_RISING,
+    /* 2 */ TAKARAYA_WALL_FALLING
+} ObjTakarayaWallSpaceState;
+
+s32 sTakarayaWallPathBuilderIndex;
+ObjTakarayaWallSpace sTakarayaWallPathBuilder[TAKARAYA_WALL_ROWS * TAKARAYA_WALL_COLUMNS];
+s32 D_80ADA508; // Set but never used
+s32 sTakarayaWallPathBuilderReachedFront;
+Vec3f sTakarayaWallAudioPositions[TAKARAYA_WALL_ROWS][TAKARAYA_WALL_COLUMNS];
+u8 sTakarayaWallSpaceStates[TAKARAYA_WALL_ROWS][TAKARAYA_WALL_COLUMNS];
+
+// -10.0f height means the wall does not raise (the space is open)
+f32 sTakarayaWallWallHeights[TAKARAYA_WALL_ROWS][TAKARAYA_WALL_COLUMNS];
+
+// Note: returns true if the space is valid
+s32 ObjTakarayaWall_PosToSpace(Vec3f* pos, s32* row, s32* column) {
+    *row = ((s32)pos->x + (14 * 120)) / 120;
+    *column = (s32)pos->z / 120;
+
+    if ((*row < 0) || (*row >= TAKARAYA_WALL_ROWS) || (*column < 0) || (*column >= TAKARAYA_WALL_COLUMNS)) {
+        return false;
+    }
+    return true;
+}
+
+/**
+ * Checks if the given row and column is valid and is open, i.e does not raise up.
+ */
+s32 ObjTakarayaWall_SpaceOpen(s32 row, s32 column) {
```

Comment:
```suggestion
s32 ObjTakarayaWall_IsCellOpen(s32 row, s32 column) {
```

I think it's much more common to refer to sections of a maze as a cell rather than space, and that references to spaces should be converted to cell.

Reply:
Agreed. Cell does make more sense.